### PR TITLE
Review/Update SAML docs to include experimental setting steps

### PR DESCRIPTION
**Pull request by justinegeffen**
Updated with content from: https://mattermost.atlassian.net/browse/MM-20862

### CODE REVIEW

File: source/deployment/sso-saml-before-you-begin.rst [hunk 1]
```diff
@@ -1,7 +1,13 @@
-Before you begin
+Before You Begin
 ----------------
 
-Before you configure SAML:
+Starting with version 5.20, a new SAML library implementation is available. This implementation does not require the installation of the XML Security Library. It is recommended that you use this new implementation as the old implementation will eventually be deprecated.
+
+To use the new implementation, navigate to **System Console > Experimental Features > Use Improved SAML Library**. Activate the feature and choose **Save** and then restart the server. The configuration change will not take effect until the server is restarted.
```
Comment (justinegeffen) [inline source/deployment/sso-saml-before-you-begin.rst]: This needs to be validated against the UI when available.

File: source/deployment/sso-saml-before-you-begin.rst [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
+
+To change the configuration file directly, edit the “UseNewSamlLibrary” setting under “ExperimentalSettings”.
```
Comment (justinegeffen) [inline source/deployment/sso-saml-before-you-begin.rst]: Change it in the config.json file?

Comment (sbishel) [reply source/deployment/sso-saml-before-you-begin.rst]: Yes.  The config.json file. And the setting is actually "UseNewSAMLLibrary". I probably had it wrong in my documentation. 

"ExperimentalSettings": {
        "ClientSideCertEnable": false,
        "ClientSideCertCheck": "secondary",
        "EnableClickToReply": false,
        "LinkMetadataTimeoutMilliseconds": 5000,
        "RestrictSystemAdmin": false,
        "UseNewSAMLLibrary": true
    },

File: source/deployment/sso-saml-before-you-begin.rst [hunk 3]
```diff
@@ -1,7 +1,13 @@
-Before you begin
+Before You Begin
 ----------------
 
-Before you configure SAML:
+Starting with Mattermost version 5.20, a new SAML library implementation is available. This implementation does not require the installation of the XML Security Library. It is recommended that you use this new implementation as the old implementation will eventually be deprecated.
+
+To use the new implementation, navigate to **System Console > Experimental Features > Use Improved SAML Library**. Activate the feature and choose **Save** and then restart the server. The configuration change will not take effect until the server is restarted.
+
+To change the configuration file directly, edit the “UseNewSamlLibrary” setting under “ExperimentalSettings”.
```
Comment (amyblais) [inline source/deployment/sso-saml-before-you-begin.rst]:
```suggestion
To change the configuration file directly, edit the ``UseNewSamlLibrary`` setting under ``ExperimentalSettings``. 
```

Comment (amyblais) [inline source/deployment/sso-saml-before-you-begin.rst]: New config settings can be documented here: https://docs.mattermost.com/administration/config-settings.html.

Comment (justinegeffen) [reply source/deployment/sso-saml-before-you-begin.rst]: Thanks, @amyblais. That's what I thought and Scott confirmed. :)

**Comment by justinegeffen** [comment]
I've updated the config settings documentation - @sbishel, can you review those changes? I've updated the documentation for clarity.

**Comment by justinegeffen** [comment]
This PR covers https://github.com/mattermost/mattermost-webapp/pull/4416.

**Comment by justinegeffen** [comment]
@sbishel, confirming that this PR does close https://github.com/mattermost/mattermost-webapp/pull/4416 and is ready to go for 5.20. Thanks! :)

File: source/administration/config-settings.rst [hunk 4]
```diff
@@ -2263,6 +2263,21 @@ SAML
...
 
+Use New SAML Library
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+**True**: Enable an updated SAML Library, which does not require the XML Security Library (xmlsec1) to be installed.
+
+**False**: Continue using the existing implementation which uses the XML Security Library (xmlsec1).
+
+.. code-block:: sh
+
+  "ExperimentalSettings": {
```
Comment (amyblais) [inline source/administration/config-settings.rst]: This section may not be correctly formatted. See below "Enable Login With SAML" setting on how config settings are normally formatted.

Comment (justinegeffen) [reply source/administration/config-settings.rst]: Thanks Amy, will do.

**Comment by sbishel** [comment]
@justinegeffen Yes, it does...Thank you

File: source/administration/config-settings.rst (hunk 4 above)
Comment (justinegeffen) [reply source/administration/config-settings.rst]: I've changed it in line with the rest of the examples. @sbishel, can you confirm that the default is ``false`` with options ``true`` and ``false``?

Comment (justinegeffen) [reply source/administration/config-settings.rst]: Just a reminder about this confirmation, @sbishel, when you have a moment. :)

File: source/administration/config-settings.rst [hunk 5]
```diff
@@ -2263,6 +2263,16 @@ SAML
 .. note::
    In line with Microsoft ADFS guidance we recommend `configuring intranet forms-based authentication for devices that do not support WIA <https://docs.microsoft.com/en-us/windows-server/identity/ad-fs/operations/configure-intranet-forms-based-authentication-for-devices-that-do-not-support-wia>`_.
 
+Use New SAML Library
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+**True**: Enable an updated SAML Library, which does not require the XML Security Library (xmlsec1) to be installed.
+
+**False**: Continue using the existing implementation which uses the XML Security Library (xmlsec1).
+
++----------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```
Comment (amyblais) [inline source/administration/config-settings.rst]: Please use "No Wrap" line wrap mode to format the table (currently the table is not rendering).

**Pull request merged**